tests/test_archs.py

```diff
@@ -0,0 +1,247 @@
+import os
+
+import unittest
+
+try:
+    from unittest import mock
+except ImportError:
+    # `Python 2` or lower than `Python 3.3` does not
+    # have the `unittest.mock` module built-in
+    import mock
+from pythonforandroid.bootstrap import Bootstrap
+from pythonforandroid.distribution import Distribution
+from pythonforandroid.recipe import Recipe
+from pythonforandroid.build import Context
+from pythonforandroid.util import BuildInterruptingException
+from pythonforandroid.archs import (
+    Arch,
+    ArchARM,
+    ArchARMv7_a,
+    ArchAarch_64,
+    Archx86,
+    Archx86_64,
+)
+
+expected_env_gcc_keys = {
+    "CFLAGS",
+    "LDFLAGS",
+    "CXXFLAGS",
+    "TOOLCHAIN_PREFIX",
+    "TOOLCHAIN_VERSION",
+    "CC",
+    "CXX",
+    "AR",
+    "RANLIB",
+    "LD",
+    "LDSHARED",
+    "STRIP",
+    "MAKE",
+    "READELF",
+    "NM",
+    "BUILDLIB_PATH",
+    "PATH",
+    "ARCH",
+    "NDK_API",
+}
+
+
+class TestArch(unittest.TestCase):
+    def setUp(self):
+        self.ctx = Context()
+        self.ctx.ndk_api = 21
+        self.ctx.android_api = 27
+        self.ctx._sdk_dir = "/opt/android/android-sdk"
+        self.ctx._ndk_dir = "/opt/android/android-ndk"
+        self.ctx.setup_dirs(os.getcwd())
+        self.ctx.bootstrap = Bootstrap().get_bootstrap("sdl2", self.ctx)
+        self.ctx.bootstrap.distribution = Distribution.get_distribution(
+            self.ctx, name="sdl2", recipes=["python3", "kivy"]
+        )
+        self.ctx.python_recipe = Recipe.get_recipe("python3", self.ctx)
+
+    def test_arch(self):
+        arch = Arch(self.ctx)
+        with self.assertRaises(AttributeError) as e1:
+            arch.__str__()
+        self.assertEqual(
+            e1.exception.args[0], "'Arch' object has no attribute 'arch'"
+        )
+        with self.assertRaises(AttributeError) as e2:
+            getattr(arch, "target")
+        self.assertEqual(
+            e2.exception.args[0], "'NoneType' object has no attribute 'split'"
+        )
+        self.assertIsNone(arch.toolchain_prefix)
+        self.assertIsNone(arch.command_prefix)
+        self.assertIsInstance(arch.include_dirs, list)
+
+    # Here we mock two functions:
+    # - `ensure_dir` because we don't want to create any directory
+    # - `find_executable` because otherwise we will
+    #   get an error when trying to find the compiler (we are setting some fake
+    #   paths for our android sdk and ndk so probably will not exist)
+    @mock.patch("pythonforandroid.archs.find_executable")
+    @mock.patch("pythonforandroid.build.ensure_dir")
+    def test_arch_arm(self, mock_ensure_dir, mock_find_executable):
```

Comment:
Minor: why not making a dedicated `class TestArchARM`. So we create a dedicated test class per class we want to test. That way we don't end up with a too big generic test class. We can still inherit from `TestArch` so we share the `setup()` if needed

Reply:
I Applied your `subclass` suggestion (slightly different from your proposal)

Thanks @AndreMiras!!!  :smile:

Reply:
Nice :+1: